monkey/agent_plugins/exploiters/smb/src/plugin.py

```diff
@@ -72,6 +74,7 @@ def run(
             servers,
             current_depth,
             remote_agent_binary_destination_path=get_agent_dst_path(host),
+            otp=self._otp_provider.get_otp(),
```

Comment:
I wonder if the command builder should get provider and use it on-demand. My concern is there may be considerable time between when this OTP is generated and a successful login occurs and the OTP may be expired by the time it's used.